source/jmap/types.d

```diff
@@ -274,94 +262,52 @@ version(SIL):
                 ["list"]);
     }
 
-
     Asdf changesRaw(string type, string sinceState, Nullable!uint maxChanges = (Nullable!uint).init, SilStruct additionalArguments = null) {
-        import std.algorithm : map;
-        import std.array : array;
         auto invocationId = "12345678";
         auto invocation = Invocation.changes(type, activeAccountId(), invocationId, sinceState, maxChanges, additionalArguments);
         auto request = JmapRequest(listCapabilities(), [invocation], null);
         return post(request);
     }
 
-    Variable changes(string type, string sinceState, Nullable!uint maxChanges = (Nullable!uint).init, SilStruct additionalArguments = null) {
-        return changesRaw(type, sinceState, maxChanges, additionalArguments).deserialize!Variable;
-    }
-
-    Asdf setRaw(string type, string ifInState = null, SilStruct create = null, SilStruct update = null, string[] destroy_ = null, SilStruct additionalArguments = null) {
-        import std.algorithm : map;
-        import std.array : array;
+    JsonAlgebraic set(string type, string ifInState = null, StringMap!JsonAlgebraic create = null, StringMap!JsonAlgebraic update = null, string[] destroy_ = null, StringMap!JsonAlgebraic additionalArguments = null) {
         auto invocationId = "12345678";
-        auto createAsdf = parseJson(serializeJson(Variable(create)));
-        auto updateAsdf = parseJson(serializeJson(Variable(update)));
-        auto invocation = Invocation.set(type, activeAccountId(), invocationId, ifInState, createAsdf, updateAsdf, destroy_, additionalArguments);
+        auto invocation = Invocation.set(type, activeAccountId(), invocationId, ifInState, create, update, destroy_, additionalArguments);
         auto request = JmapRequest(listCapabilities(), [invocation], null);
         return post(request);
     }
 
-    Variable set(string type, string ifInState = null, SilStruct create = null, SilStruct update = null, string[] destroy_ = null, SilStruct additionalArguments = null) {
-        return setRaw(type, ifInState, create, update, destroy_, additionalArguments).deserialize!Variable;
-    }
-
     Variable setEmail(string ifInState = null, SilStruct create = null, SilStruct update = null, string[] destroy_ = null, SilStruct additionalArguments = null) {
```

Comment:
Haha, no! `private alias Variable = JsonAlgebraic;`

I did my best to minimize the MR.

Reply:
Ouch. That is not the way to go about minimizing an MR.

It would have been a lot better if first there was a MR to remove Asdf, then one to deprecate the xyzRaw stuff, then one to remove the SIL dependency, etc.

I bet if this MR only removed asdf and replaced with ion, it would have been way smaller and merged 7 days ago.

Reply:
> to remove Asdf, then one to deprecate the xyzRaw 

How to do it if Raw returns Asdf?